samcli/commands/_utils/custom_options/hook_package_id_option.py

```diff
@@ -74,15 +74,26 @@ def handle_parse_result(self, ctx, opts, args):
                 LOG.debug("beta-feature flag is disabled and prepare hook is not run")
                 return super().handle_parse_result(ctx, opts, args)
 
+            iac_project_path = os.getcwd()
+            output_dir_path = os.path.join(iac_project_path, ".aws-sam-iacs", "iacs_metadata")
+
+            # check if user provided --skip-prepare-iac
+            skip_prepare_hook = opts.get("skip_prepare_iac")
+            metadata_file_path = os.path.join(output_dir_path, "template.json")
+
             # call prepare hook
             built_template_path = DEFAULT_BUILT_TEMPLATE_PATH
             if not self._force_prepare and os.path.exists(built_template_path):
                 LOG.info("Skip Running Prepare hook. The current application is already prepared.")
+            elif skip_prepare_hook:
+                if os.path.exists(metadata_file_path):
+                    LOG.info("Skipping prepare hook, --skip-prepare-iac was provided and metadata file already exists.")
+                    opts["template_file"] = metadata_file_path
+                else:
+                    raise click.UsageError("No metadata file found, rerun without the --skip-prepare-iac flag.")
```

Comment:
This needs to be more expressive, what is a metadata file, how do customers know how to generate it.

On the error message itself, should we instead have the command to be run?

eg: Try: ` sam build --skip-prepare-iac`

Reply:
New behaviour is as follows:
- if option is provided and metadata file exists: console log a message saying we skipped prepare since we found that file at `/path/here`
- if option if not provided, or if option is provided and metadata file does not exist: continue with execution without printing any message

Sample output when using `--skip-prepare-infra`:
```
Running Prepare Hook to prepare the current application
Executing prepare hook of hook package "terraform"
Skipping preparation stage, the metadata file already exists at /Users/lucashuy/repos/Aws-sam-cli-terraform-sample-projects/working_examples/needs_build_nodejs/.aws-sam-iacs/iacs_metadata/template.json
Prepare hook completed and metadata file generated at: /Users/lucashuy/repos/Aws-sam-cli-terraform-sample-projects/working_examples/needs_build_nodejs/.aws-sam-iacs/iacs_metadata/template.json
Building layer 'aws_lambda_layer_version.hello_world_layer'
```